lightning/src/events/mod.rs

```diff
@@ -476,8 +473,13 @@ pub enum Event {
 		/// The purpose of the claimed payment, i.e. whether the payment was for an invoice or a
 		/// spontaneous payment.
 		purpose: PaymentPurpose,
-		/// The HTLCs that comprise the claimed payment.
+		/// The HTLCs that comprise the claimed payment. This will be empty for events serialized prior
+		/// to LDK version 0.0.117.
 		htlcs: Vec<ClaimedHTLC>,
```

Comment:
Shouldn't we make this an `Option<Vec<ClaimedHTLC>>`? I'm thinking it will help us better signal the absence of the field prior to 0.0.117.

Reply:
We could, although I'm not sure it would add much value. In particular, any claimed payment _must have at least one HTLC_, so any `PaymentClaimed` where this vector is empty must necessarily be pre-0.117. But I'm happy to make the folks think it's clearer / more consistent.

Reply:
I think we can leave as is due to it being necessarily non-empty with 117+. It at least removes the one layer of `Option` indirection. Otherwise I'm pretty indifferent to it, lol